File: 351004/Shkredova/src/main/java/com/example/task310/model/Notice.java
package com.example.task310.model;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;
import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "tbl_notice")
public class Notice {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 2048)
    private String content;

    @CreationTimestamp
    @Column(updatable = false)
    private LocalDateTime created;

    @UpdateTimestamp
    private LocalDateTime modified;

    @ManyToOne
    @JoinColumn(name = "news_id", nullable = false)
    private News news;
}

File: 351003/brezgunov/distcomp/src/main/kotlin/com/distcomp/entity/Notice.kt
package com.distcomp.entity

import jakarta.persistence.*

@Entity
@Table(name = "tbl_notice")
class Notice(
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    var id: Long,
    var content: String,
    @ManyToOne
    @JoinColumn(name = "news_id")
    var news: News? = null
)